src/forms/renderers/Overrides/material/complex/MaterialoneOfRenderer_Format.tsx

```diff
@@ -0,0 +1,209 @@
+  
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import {
+    CombinatorRendererProps,
+    createCombinatorRenderInfos,
+    createDefaultValue,
+    JsonSchema,
+    OwnPropsOfControl,
+    RankedTester,
+    rankWith,
+    scopeEndsWith,
+} from '@jsonforms/core';
+import { JsonFormsDispatch, withJsonFormsOneOfProps } from '@jsonforms/react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    Hidden,
+    Tab,
+    Tabs,
+} from '@mui/material';
+import { has } from 'lodash';
+import isEmpty from 'lodash/isEmpty';
+import { useCallback, useState } from 'react';
+import CombinatorProperties from './CombinatorProperties';
+
+export interface OwnOneOfProps extends OwnPropsOfControl {
+    indexOfFittingSchema?: number;
+}
+
+export const Custom_MaterialOneOfRenderer = ({
+    handleChange,
+    schema,
+    path,
+    renderers,
+    cells,
+    rootSchema,
+    id,
+    visible,
+    indexOfFittingSchema,
+    uischema,
+    uischemas,
+    data,
+}: CombinatorRendererProps) => {
+    const [open, setOpen] = useState(false);
+    const [selectedIndex, setSelectedIndex] = useState(
+        indexOfFittingSchema || 0
+    );
+    const [newSelectedIndex, setNewSelectedIndex] = useState(0);
+    const handleClose = useCallback(() => setOpen(false), [setOpen]);
+    const cancel = useCallback(() => {
+        setOpen(false);
+    }, [setOpen]);
+
+    // const descriminator = schemaClone.discriminator?.propertyName ?? null;
+
+    const oneOfRenderInfos = createCombinatorRenderInfos(
+        (schema as JsonSchema).oneOf,
+        rootSchema,
+        'oneOf',
+        uischema,
+        path,
+        uischemas
+    );
+
+    // Run through the elements and clear out the ones without elements
+    oneOfRenderInfos.map((renderer) => {
+        const { uischema: rendererUischema } = renderer as any;
+        rendererUischema.elements = rendererUischema.elements.filter(
+            (el: any) => el !== null
+        );
+        return renderer;
+    });
+
+    const openNewTab = (newIndex: number) => {
+        const tabSchema = oneOfRenderInfos[newIndex].schema;
+        const { properties: tabSchemaProps } = tabSchema;
+        const defaultVal: {
+            formatter: string;
+            settings?: any;
+        } = {
+            formatter: tabSchemaProps?.formatter.default ?? '',
+        };
+
+        if (has(tabSchemaProps, 'settings')) {
+            defaultVal.settings = {};
+        }
```

Comment:
Customization 2: We want to control how the default val is set here as JSONForms and AJV do not handle every case possible when using `oneOf`s.